share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md

```diff
@@ -0,0 +1,42 @@
+# How to Create and Maintain a Tap
+Taps are external sources of Homebrew formulae. They can be created by anyone
+to provide their own formulae to any Homebrew user.
+
+## Creating a tap
+A tap must be a git repository available online. It can be hosted on GitHub or
+elsewhere. If hosted on GitHub, the repo’s name must start with `homebrew-`.
+
+Formulae follow the same format as the core’s ones, and can be added under
+`Formula` or `HomebrewFormula`. It works if they are at the root of the repo
+but this is not recommended.
```

Comment:
Most taps will be small and will only contain formulae. It's simpler and (conversely to your third point) makes more visible which formulae a tap provides if the formulae are placed in the root. My preference is to advise new tap owners to put formulae in the repository root, and then say that formulae can also be placed inside a `Formula` or `HomebrewFormula` directory.